src/test/java/org/folio/rest/api/ItemEffectiveCallNumberComponentsTest.java

```diff
@@ -0,0 +1,294 @@
+package org.folio.rest.api;
+
+
+import static org.apache.commons.lang3.StringUtils.capitalize;
+import static org.folio.rest.api.ItemStorageTest.nod;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.builders.HoldingRequestBuilder;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+@RunWith(JUnitParamsRunner.class)
+public class ItemEffectiveCallNumberComponentsTest extends TestBaseWithInventoryUtil {
+  private static final String HOLDINGS_CALL_NUMBER_TYPE = UUID.randomUUID().toString();
+  private static final String HOLDINGS_CALL_NUMBER_TYPE_SECOND = UUID.randomUUID().toString();
+  private static final String ITEM_LEVEL_CALL_NUMBER_TYPE = UUID.randomUUID().toString();
+  private static final String ITEM_LEVEL_CALL_NUMBER_TYPE_SECOND = UUID.randomUUID().toString();
+
+  @BeforeClass
+  public static void createCallNumberTypes() throws Exception {
```

Comment:
What happens if these records already exist?

Reply:
I don't think it is possible, because the record will be created only once, and will have a random ID.

Reply:
Are there any constraints on the name of the call number type?

My concern is that these records aren't cleaned up between tests.

Reply:
Yes, we have a UK for the name field. I'm specifying a test-specific name, I don't think that we have the same name. Alternatively, I can re-use existing types, from reference data. Does it sound better?

Reply:
I’m not suggesting reusing records from the reference data. I’d prefer the tests didn’t do that because I think they should be independent of it.

My concern is that these records don’t appear to be removed. Are they removed elsewhere in a general cleanup?

Reply:
I don't think, I can remove it in `@AfterClass` method.

Reply:
I typically suggest cleaning up records before tests run rather than after, in order to aid diagnostics if tests fail.